src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -26287,15 +26384,25 @@ public class XAttribute : System.Attribute
                 """;
 
             CreateCompilation(source).VerifyEmitDiagnostics(
-                // (1,4): error CS1503: Argument 1: cannot convert from 'collection expressions' to 'ERROR[]'
-                // [X([1])]
-                Diagnostic(ErrorCode.ERR_BadArgType, "[1]").WithArguments("1", "collection expressions", "ERROR[]").WithLocation(1, 4),
                 // (8,23): error CS0246: The type or namespace name 'ERROR' could not be found (are you missing a using directive or an assembly reference?)
                 //     public XAttribute(ERROR[] values) { }
                 Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "ERROR").WithArguments("ERROR").WithLocation(8, 23)
                 );
         }
 
+        [Fact]
+        public void InAssignment_BadArrayType()
+        {
+            string source = """
+                ERROR[] values = [1];
+                values = [2];
```

Comment:
That's interesting, and seems reasonable I guess, as an "avoid cascading" behavior? Is this similar to how `values = new[] { 2 }` would behave?

Reply:
Yes, the behavior is similalr for `values = new[] { 2 }` which doesn't report any error (since it is assumed we've already reported an error on the declaration of `ERROR[] values;`.

However, the error that is suppressed in `values = new[] { 2 }` is converting `int[]` to `ERROR[]`, where as with `values = [2]` the error that is suppressed is converting `int` to `ERROR`.

Also, the error reporting for assignment did not change in this PR - I added this test to show the similar diagnostics to the previous test `InAttribute_BadArrayType()` that was affected by this PR. In that test, a call to `[X(new[] { 2 })]` will result in a conversion error for the argument.

I'm on the fence about whether we should report an error at the for `[X([1])]` in that test.

Reply:
> However, the error that is suppressed in `values = new[] { 2 }` is converting `int[]` to `ERROR[]`, whereas with `values = [2]` the error that is suppressed is converting `int` to `ERROR`.

Another example is `values = default` which does not result in any additional error.

Reply:
that all sounds good, merge at will :)